src/Itinero/Router.cs

```diff
@@ -1320,13 +1320,24 @@ public sealed override Result<bool> TryCheckConnectivity(IProfileInstance profil
                         }
                         weights = algorithm.Weights;
                     }
-                    //else if (contracted.HasNodeBasedGraph &&
-                    //   contracted.NodeBasedIsEdgedBased)
-                    //{ // use vertex-based graph for edge-based routing.
-                    //    weights = Itinero.Algorithms.Contracted.Dual.RouterExtensions.CalculateManyToMany(contracted,
-                    //        _db, profileInstance.Profile,
-                    //        weightHandler, sources, targets, maxSearch, cancellationToken).Value;
-                    //}
+                    else if (contracted.HasNodeBasedGraph &&
+                       contracted.NodeBasedIsEdgedBased)
+                    { // use vertex-based graph for edge-based routing.
+                      //weights = Itinero.Algorithms.Contracted.Dual.RouterExtensions.CalculateManyToMany(contracted,
+                      //    _db, profileInstance.Profile,
+                      //    weightHandler, sources, targets, maxSearch, cancellationToken).Value;
+
+                        var algorithm = new Itinero.Algorithms.Default.ManyToMany<float>(_db, weightHandler, sources, targets, maxSearch);
```

Comment:
As you can see from the commented code above CalculateManyToMany was previously being used but returned different results which didn't match that of he non contracted versions. This change fixed both the simple and complex cases I was made aware of.